src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleBinaryOperator.cs

```diff
@@ -79,13 +82,13 @@ private bool IsLikeTupleExpression(BoundExpression expr, out BoundTupleExpressio
                         }
                         var newArguments = builder.ToImmutableAndFree();
                         tuple = new BoundConvertedTupleLiteral(
-                            tuple.Syntax, sourceTuple: null, wasTargetTyped: true, newArguments, ImmutableArray<string>.Empty,
+                            tuple.Syntax, sourceTuple: null, wasTargetTyped: true, newArguments, ImmutableArray<string?>.Empty,
                             ImmutableArray<bool>.Empty, conversion.Type, conversion.HasErrors);
                         return true;
                     }
                 case BoundConversion { Conversion: { Kind: var kind }, Operand: var o } when
                         (kind == ConversionKind.ImplicitNullable || kind == ConversionKind.ExplicitNullable) &&
-                        expr.Type.IsNullableType() && expr.Type.StrippedType().Equals(o.Type, TypeCompareKind.AllIgnoreOptions):
+                        expr.Type!.IsNullableType() && expr.Type!.StrippedType().Equals(o.Type, TypeCompareKind.AllIgnoreOptions):
```

Comment:
Consider pulling `Type` out of the conversion with a pattern, instead of suppressing.